dot/network/light.go

```diff
@@ -0,0 +1,500 @@
+	"github.com/ChainSafe/gossamer/lib/scale"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// Pair is a pair of arbitrary bytes.
+type Pair struct {
+	first  []byte
+	second []byte
+}
+
+// LightRequest is all possible light client related requests.
+type LightRequest struct {
+	RmtCallRequest      RemoteCallRequest
+	RmtReadRequest      RemoteReadRequest
+	RmtHeaderRequest    RemoteHeaderRequest
+	RmtReadChildRequest RemoteReadChildRequest
+	RmtChangesRequest   RemoteChangesRequest
+}
+
+// LightResponse is all possible light client response messages.
+type LightResponse struct {
+	RmtCallResponse   RemoteCallResponse
+	RmtReadResponse   RemoteReadResponse
+	RmtHeaderResponse RemoteHeaderResponse
+	RmtChangeResponse RemoteChangesResponse
+}
+
+// RemoteCallRequest ...
+type RemoteCallRequest struct {
+	Block  []byte
+	Method string
+	Data   []byte
+}
+
+// RemoteReadRequest ...
+type RemoteReadRequest struct {
+	Block []byte
+	Keys  [][]byte
+}
+
+// RemoteReadChildRequest ...
+type RemoteReadChildRequest struct {
+	Block      []byte
+	StorageKey []byte
+	Keys       [][]byte
+}
+
+// RemoteHeaderRequest ...
+type RemoteHeaderRequest struct {
+	Block []byte
+}
+
+// RemoteChangesRequest ...
+type RemoteChangesRequest struct {
+	FirstBlock *optional.Hash
+	LastBlock  *optional.Hash
+	Min        []byte
+	Max        []byte
+	StorageKey *optional.Bytes
+	key        []byte
+}
+
+// RemoteCallResponse ...
+type RemoteCallResponse struct {
+	Proof []byte
+}
+
+// RemoteReadResponse ...
+type RemoteReadResponse struct {
+	Proof []byte
+}
+
+// RemoteHeaderResponse ...
+type RemoteHeaderResponse struct {
+	Header []*optional.Header
+	proof  []byte
+}
+
+// RemoteChangesResponse ...
+type RemoteChangesResponse struct {
+	Max        []byte
+	Proof      [][]byte
+	Roots      [][]Pair
+	RootsProof []byte
+}
+
+// String formats a RemoteCallRequest as a string
+func (rc *RemoteCallRequest) String() string {
+	return fmt.Sprintf("Block =%s method=%s Data=%s",
+		string(rc.Block), rc.Method, string(rc.Data))
+}
+
+// Encode encodes a RemoteCallRequest message using SCALE and appends the type byte to the start
+func (rc *RemoteCallRequest) Encode() ([]byte, error) {
+	enc, err := scale.Encode(rc)
+	if err != nil {
+		return enc, err
+	}
+	return append([]byte{RemoteCallRequestType}, enc...), nil
+}
+
+// Decode the message into a RemoteCallRequest, it assumes the type byte has been removed
+func (rc *RemoteCallRequest) Decode(r io.Reader) error {
+	sd := scale.Decoder{Reader: r}
+	_, err := sd.Decode(rc)
+	return err
+}
+
+// Type returns RemoteCallRequestType
+func (rc *RemoteCallRequest) Type() int {
+	return RemoteCallRequestType
+}
+
+// IDString Returns an empty string to ensure we don't rebroadcast it
+func (rc *RemoteCallRequest) IDString() string {
+	// scale encode each extrinsic
+	encMsg, err := rc.Encode()
+	if err != nil {
+		return ""
+	}
+	hash, err := common.Blake2bHash(encMsg)
+	if err != nil {
+		return ""
+	}
+	return hash.String()
+}
```

Comment:
this is going to be deprecated so you can just return an empty string, same for all the others, sorry I didn't mention it